subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ecdsa.c

```diff
@@ -0,0 +1,554 @@
+
+		status = deterministic_ecdsa_hmac(hashalg, K, V, digestsz, INTERNAL_OCTET_ZERO,
+						  privkey->d, T, opsz, K);
+		break;
+
+	case 1: /* V = HMAC_K(V) */
+		status = deterministic_ecdsa_hmac(hashalg, K, V, digestsz, INTERNAL_OCTET_NOT_USED,
+						  NULL, NULL, opsz, V);
+		break;
+
+	case 2: /* K = HMAC_K(V || 0x01 || privkey || h1) */
+		status = deterministic_ecdsa_hmac(hashalg, K, V, digestsz, INTERNAL_OCTET_ONE,
+						  privkey->d, T, opsz, K);
+		break;
+
+	case 3: /* V = HMAC_K(V) */
+	case 4: /* same as case 3. */
+		status = deterministic_ecdsa_hmac(hashalg, K, V, digestsz, INTERNAL_OCTET_NOT_USED,
+						  NULL, NULL, opsz, V);
+		break;
+
+	case 5: /* T = T || V */
+		size_t copylen = MIN(digestsz, opsz - hmac_op->tlen);
+
+		memcpy(T + hmac_op->tlen, V, copylen);
+		hmac_op->tlen += copylen;
+		if (hmac_op->tlen < opsz) {
+			/* We need more data. */
+			hmac_op->step = 4;
+			return SX_ERR_HW_PROCESSING;
+		}
+		break;
+
+	case 6: /* Verify that T is in range [1, q-1] */
+	{
+		int rshift = clz_u8(curve_n[0]);
+
+		/* Only use the first qlen bits that was generated. */
+		bits2int(T, opsz, T, opsz * 8 - rshift);
+
+		bool is_zero = true;
+
+		for (size_t i = 0; i < opsz; i++) {
+			if (T[i]) {
+				is_zero = false;
+				break;
+			}
+		}
+		int ge = be_cmp(T, curve_n, opsz, 0);
+		bool must_retry =
+			hmac_op->deterministic_retries < (MAX_ECDSA_ATTEMPTS - hmac_op->attempts);
+		if (is_zero || ge >= 0 || must_retry) {
+			/* T is out of range. Retry */
+			hmac_op->step = 3;
+			hmac_op->tlen = 0;
+			hmac_op->deterministic_retries++;
+			/* K = HMAC_K(V || 0x00) */
+			deterministic_ecdsa_hmac(hashalg, K, V, digestsz, INTERNAL_OCTET_ZERO, NULL,
+						 NULL, 0, K);
+			return SX_ERR_HW_PROCESSING;
+		}
+		memcpy(workmem, h1, digestsz);
+		memcpy(workmem + digestsz, T, opsz);
+		break;
+	}
+	default:
+		status = SX_ERR_INVALID_PARAM;
+	}
+	hmac_op->step++;
+	return status;
+}
+
+static int deterministic_ecdsa_hmac(const struct sxhashalg *hashalg, const uint8_t *key,
+				    const uint8_t *v, size_t hash_len, uint8_t internal_octet,
+				    const uint8_t *sk, const uint8_t *hash, size_t key_len,
+				    uint8_t *hmac)
+{
+
+	struct sxhash hashopctx;
+	size_t workmemsz = sx_hash_get_alg_digestsz(hashalg) + sx_hash_get_alg_blocksz(hashalg);
+	char workmem[workmemsz];
+	int status;
+
+	status = mac_create_hmac(hashalg, &hashopctx, key, hash_len, workmem, workmemsz);
+	if (status != SX_OK) {
+		return status;
+	}
+	/* The hash function is initialized in mac_create_hmac */
+	status = sx_hash_feed(&hashopctx, v, hash_len);
+	if (status != SX_OK) {
+		return status;
+	}
+
+	if (internal_octet != INTERNAL_OCTET_NOT_USED) {
+
+		static uint8_t internal_octet_values[] = {
+			[INTERNAL_OCTET_ZERO] = 0,
+			[INTERNAL_OCTET_ONE] = 1,
+			[INTERNAL_OCTET_UNUSED] = 0xFF,
+
+		};
+		uint8_t value = internal_octet_values[internal_octet];
+
+		status = sx_hash_feed(&hashopctx, &value, sizeof(value));
```

Comment:
Is it intended to ignore the sx_hash_feed result in this call and in the next two ones?

Reply:
Yes, because you check status at 493, and for some reason I believed that these were mutually exclusive, but they are not. I will update